src/main/webapp/app/exercises/shared/plagiarism/plagiarism-split-view/split-pane-header/split-pane-header.component.ts

```diff
@@ -45,19 +128,61 @@ export class SplitPaneHeaderComponent implements OnChanges {
         return this.files[this.activeFileIndex].file;
     }
 
+    /**
+     * handles selection of file from dropdown, propagates change to fileslectionsubject component for lock sync
+     * @param file to be selected
+     * @param idx index of the file from the dropdown
+     */
     handleFileSelect(file: FileWithHasMatch, idx: number): void {
+        this.fileSelectedSubject()!.next({ idx: idx, file: file });
+        this.activeFileIndex = idx;
+        this.showFiles = false;
+        this.selectFile.emit(file.file);
+    }
+
+    /**
+     * handles selection of file from dropdown, do NOT propagates change to fileslectionsubject component for lock sync
+     * @param file to be selected
+     * @param idx index of the file from the dropdown
+     */
+    handleFileSelectWithoutPropagation(file: FileWithHasMatch, idx: number) {
         this.activeFileIndex = idx;
         this.showFiles = false;
         this.selectFile.emit(file.file);
+        file.hasMatch = true;
     }
 
     hasFiles(): boolean {
-        return this.files && this.files.length > 0;
+        return (this.files && this.files.length > 0) ?? false;
     }
 
     toggleShowFiles(): void {
```

Comment:
you could also combine `toggleShowFiles` and `toggleShowFilesWithoutPropagation `